File: go/hamming/hamming.go
package hamming

import "fmt"

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, fmt.Errorf("Strands must be the same length")
	}
	var distance int
	for n, c := range []byte(a) {
		if c != b[n] {
			distance++
		}
	}
	return distance, nil
}

File: ruby/hamming/hamming.rb
module Hamming

  def self.compute(cad1, cad2)
    if cad1.length != cad2.length
      raise ArgumentError.new("Strands must be the same length")
    else
      cad1.chars.zip(cad2.chars).count { |a, b| a!=b }
    end
  end
end